fedmsg_meta_fedora_infrastructure/fasshim.py

```diff
@@ -113,15 +113,13 @@ def make_fasjson_cache(**config):
         log.error("Something went wrong setting up fasjson client with error: %s" % e)
         return {}
     try:
-        response = client.list_users(page_size=1000)
-        _add_to_cache(response.result)
-        total_pages = response.page.get('total_pages')
-        # we want to start at the second page, since we already have the first
-        page_number = 2
-        while page_number < total_pages+1:
-            users = client.list_users(page_size=1000, page=page_number).result
-            _add_to_cache(users)
-            page_number += 1
+        page_number = 0
+        next_page_exists = True
+        while next_page_exists:
+            page_number +=1
+            response = client.list_users(page_size=1000, page=page_number)
+            _add_to_cache(response.result)
+            next_page_exists = page_number < response.page.get("total_pages")
```

Comment:
Do you envision a case where `total_pages` will not be present in the page dict? If you know it's there you can just do `page["total_pages"]`. Using the `get` accessor is only useful if you want the call to return None if the key is absent (which would also crash the code here BTW, since comparing a number and None will raise an exception) or return a default value.

Reply:
Good point, I'll make that change.